src/extrovermectin.ts

```diff
@@ -0,0 +1,85 @@
+import { useFamiliar } from "kolmafia";
+import {
+  $item,
+  $items,
+  $location,
+  $monster,
+  $skill,
+  adventureMacro,
+  clamp,
+  get,
+  have,
+  Requirement,
+} from "libram";
+import { freeFightFamiliar } from "./familiar";
+import { findRun, ltbRun } from "./lib";
+import { Macro } from "./combat";
+
+export function expectedGregs(): number {
+  const baseGregs = 3;
+  const timeSpunGregs = have($item`Time-Spinner`)
+    ? Math.floor((10 - get("_timeSpinnerMinutesUsed")) / 3)
+    : 0;
+  const orbGregs = have($item`miniature crystal ball`) ? 1 : 0;
+
+  const macrometeors = have($skill`Meteor Lore`) ? 10 - get("_macrometeoriteUses") : 0;
+  const replaceEnemies = have($item`Powerful Glove`)
+    ? Math.floor((100 - get("_powerfulGloveBatteryPowerUsed")) / 10)
+    : 0;
+  const totalMonsterReplacers = macrometeors + replaceEnemies;
+
+  const sabersLeft = have($item`Fourth of May Cosplay Saber`)
+    ? clamp(5 - get("_saberForceUses"), 0, 3)
+    : 0;
+
+  const baseRateMultiplier = have($skill`Transcendent Olfaction`) ? 0.95 : 0.75;
+  const monsterReplacerGregs = clamp(
+    totalMonsterReplacers,
+    0,
+    2 * sabersLeft + baseRateMultiplier * (totalMonsterReplacers - 2 * sabersLeft)
+  );
+  const gregs = baseGregs + timeSpunGregs + orbGregs + monsterReplacerGregs;
+  return gregs;
+}
+
+export function doingExtrovermectin(): boolean {
+  return get("beGregariousCharges") > 0;
+}
+
+export function crateStrategy(): "Sniff" | "Saber" | "Orb" | null {
+  if (!doingExtrovermectin()) return null;
+  if (have($skill`Transcendent Olfaction`)) return "Sniff";
+  if (have($item`miniature crystal ball`)) return "Orb";
```

Comment:
Do we also want a strategy that uses both olfaction and orb if possible, or is that too marginal to be useful? On round 1, the embez isn't in the queue yet, so it is 1/4 to hit your first greg there. It gets a lot less likely (1/13) with the orb and olfaction.
Unless I'm misunderstanding what this function does. The way I read this is that it is there to say how we want to ensure that we hit crates and not gregs so we can hit all 20 macro-ed Embezs

Reply:
You're reading it right. Right now, our code uses orb regardless of whether we actually have olfaction. I hadn't planned out all of the code as I wrote this function.

Reply:
That being said, I'd like to keep it as is until I'm more certain of how much the rest of the code is going to change.

Reply:
Oh I see. Since we are going to orb if possible anyway, Sniff means "sniff and if possible orb". Orb means "orb only" and if we have neither, we have to use saber.
That makes sense.